### PR TITLE
[cloudflare] add tool to clear cache

### PR DESCRIPTION
**Pull request by tuannvm**
## Why?
- Tool to manage cloudflare resources
- Add drone pipeline 
- Push docker image to https://quay.io/repository/honestbee/devops-tools

### CODE REVIEW

File: cloudflarectl/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# CloudflareCTL
```
Comment (joerx) [inline cloudflarectl/README.md]: Markdown: VSCode has a markdown linter built in - please check its suggestions, there are a few formatting issues here.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+pipeline:
```
Comment (joerx) [inline .drone.yml]: Are we going to have a pipeline step for each tool we're building here? Means each change would rebuild all images? Seems rather excessive to me.

Is this tool in any way secret? If not, can we just create a separate (public) repo and a Travis build for it?

File: cloudflarectl/README.md (hunk 1 above)
Comment (joerx) [reply cloudflarectl/README.md]: Newline after heading

File: cloudflarectl/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+A tool to help manage cloudflare resources
+
+## Installation
+- Binary could be downloaded [here](tech.honestbee.com/devops-tools/cloudflarectl.tar.gz)
```
Comment (joerx) [inline cloudflarectl/README.md]: Newline before and after bullet list

File: cloudflarectl/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+```bash
```
Comment (joerx) [inline cloudflarectl/README.md]: Indent code block by one level to align it with the bullet points (or remove the bullet list up there)

File: cloudflarectl/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+curl tech.honestbee.com/devops-tools/cloudflarectl.tar.gz | tar xvf
+chmod +x cloudflarectl
+mv cloudflarectl /usr/local/bin/
+```
+
+- Docker version is available [here](https://quay.io/repository/honestbee/devops-tools?tab=tags)
+```
```
Comment (joerx) [inline cloudflarectl/README.md]: Language mode missing

File: .drone.yml (hunk 2 above)
Comment (tuannvm) [reply .drone.yml]: > Means each change would rebuild all images? Seems rather excessive to me.

We can personally decide which tool to be built then, as I specified the name `build-cloudflarectl`

> Is this tool in any way secret? If not, can we just create a separate (public) repo and a Travis build for it?

Nope, it's public-access tool. Using Travis will add more overhead then, as we need to maintain another format style, I would love to use `Drone cloud` if it's available.

File: cloudflarectl/README.md (hunk 1 above)
Comment (tuannvm) [reply cloudflarectl/README.md]: comment addressed

File: .drone.yml (hunk 2 above)
Comment (joerx) [reply .drone.yml]: @tuannvm

> We can personally decide which tool to be built then, as I specified the name build-cloudflarectl

How? Each push to master will rebuild all the images, even the ones with no changes. 

How to pin image versions? Will we have a global version number on the repo? Then each new tag will release new images, but again, with no changes

I think this repo is more suited as a scratch space to drop random scripts used for administrative tasks.

This particular tool seems to be a standalone app, so for better management I'd prefer to have it in its own repo.

Point taken for drone.yml - the build seems completely docker based, can't we just use Quay.io with build triggers?

@so0k what do you think?

Comment (tuannvm) [reply .drone.yml]: > How? Each push to master will rebuild all the images, even the ones with no changes.
How to pin image versions? Will we have a global version number on the repo? Then each new tag will release new images, but again, with no changes

Ah you mean updating one tool will affect the others as well. I think we could filter out the specific tool needed to build via `drone tag`:

```yaml
when:
  event: tag
  branch: refs/tags/cloudflarectl-*
```

> Point taken for drone.yml - the build seems completely docker based, can't we just use Quay.io with build triggers?

As I'm afraid we need to configure build trigger for each of repositories, which is quite tedious if we have many more tools

**Comment by so0k** [comment]
we have an existing ticket and discussion thread for the CI/CD set up the Repo, this is a separate concern and should not be part of this PR.

File: .drone.yml [hunk 6]
```diff
@@ -1,24 +0,0 @@
-pipeline:
```
Comment (tuannvm) [inline .drone.yml]: @so0k @joerx drone removed

File: cloudflarectl/README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+```bash
+NAME:
+   cloudflarectl - golang tool to clear cloudflare cache
+
+USAGE:
+   cloudflarectl [global options] command [command options] [arguments...]
+
+VERSION:
+   0.9.0
```
Comment (so0k) [inline cloudflarectl/README.md]: Terraform version is 0.11 after 3 years ;)

**Comment by so0k** [comment]
Did you look at [flarectl](https://github.com/cloudflare/cloudflare-go/tree/master/cmd/flarectl) from CloudFlare themselves?

Also, note that existing CI/CD was using [cli4](https://github.com/cloudflare/python-cloudflare#cli)

**Comment by tuannvm** [comment]
Looks like I'm reinventing the wheel :D I guess for this kind of tool I would keep it in my personal account then.

**Comment by tuannvm** [comment]
Close for now

**Pull request closed without merging**